feat/catalog-search-field: Add search field support

ayaanqui opened the pull request:
### Features to implement

- [x] Add "All" badge for "All Available Strains"
  - [x] Update tests
- [x] Descriptor autocomplete
- [x] Check GraphQL API implementation for descriptor

ayaanqui committed:
feat: change ActiveFilter type from string[] to Filter[]

ayaanqui committed:
feat: Show "All" badge for "All Available Strains" option

ayaanqui committed:
test: update test cases to include "All" badge

ayaanqui committed:
test: use Filter type instead of string

ayaanqui committed:
test: remove redundant clear button test

ayaanqui committed:
feat: implement search drop down

ayaanqui committed:
feat: implement search drop down

ayaanqui committed:
test: add test cases for search fields and clear button

cybersiddhu reviewed (changes requested):
* The search do not search anything.

cybersiddhu reviewed (changes requested):
Great work 👏 so far.
Just disregard my previous comment, there are couple of tweaks that needed to be done and i will  explained it visually.

ayaanqui committed:
refactor: using useCatalogDispatch

ayaanqui committed:
feat: update strainInitialState to "none"

ayaanqui committed:
feat: select field value from context searchBoxDropdownValue

ayaanqui committed:
feat: add field and search query params

ayaanqui committed:
feat: add field and search query params

ayaanqui committed:
feat: append field query if present

ayaanqui committed:
feat: append field query if present

ayaanqui committed:
feat: use dropdown value from field query

ayaanqui committed:
feat: use dropdown value from field query

ayaanqui committed:
feat: add function to update queries with state

ayaanqui committed:
feat: add function to update queries with state

ayaanqui committed:
fix: properly update queries with state

ayaanqui committed:
fix: properly update queries with state

ayaanqui committed:
feat: use updateSearchQueries to remove filters

ayaanqui committed:
feat: reset searchValue and dropdownValue if updateDropdown param is none

ayaanqui committed:
fix: dispatch setQueryVaraibles and setSearchValue if search query is provided

ayaanqui committed:
feat: update chips and persist state

ayaanqui committed:
feat: update chips and persist state

ayaanqui committed:
fix: undefined event.target error on empty input submit

ayaanqui committed:
test: fix existing tests

ayaanqui committed:
feat: use updateSearchQueries instead of hard-coded URI

ayaanqui committed:
fix: remove "No options" dropdown message

ayaanqui committed:
feat: disable summary and id fields temporarily

ayaanqui committed:
refactor: create separate useEffect for filter, search and field queries

ayaanqui committed:
test: remove test to open field dropdown as it does not exist

ayaanqui committed:
feat: attempt#1 at recreating GCloud filter UI

ayaanqui committed:
fix: pass visual tests

ayaanqui committed:
refactor: move search check with updateData in useEffect

ayaanqui committed:
fix: reset hasMore to fix infinite scroll bug

dependabot[bot] committed:
chore(deps): bump react-router-dom from 5.3.0 to 6.0.1

Bumps [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom) from 5.3.0 to 6.0.1.
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Commits](https://github.com/remix-run/react-router/commits/v6.0.1/packages/react-router-dom)

---
updated-dependencies:
- dependency-name: react-router-dom
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
chore(deps): bump react-router-dom from 5.3.0 to 6.0.1

Bumps [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom) from 5.3.0 to 6.0.1.
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Commits](https://github.com/remix-run/react-router/commits/v6.0.1/packages/react-router-dom)

---
updated-dependencies:
- dependency-name: react-router-dom
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

ayaanqui committed:
fix: replace Redirect with Navigate

ayaanqui committed:
fix: remove unsupported generic type declaration Params

ayaanqui committed:
fix: replace useHistory with useNavigate

ayaanqui committed:
fix: replace useHistory with useNavigate

ayaanqui committed:
fix: replace Redirect with Navigate OrderConfirmation

ayaanqui committed:
fix: remove unsupported generic type declaration Params

ayaanqui committed:
fix: remove useHistory.push references

ayaanqui committed:
fix: remove exact prop + replace component with element prop

ayaanqui committed:
fix: nest routes with react-router v6

ayaanqui committed:
fix: nested routes with index for parent route

ayaanqui committed:
fix: PrivateRoute returns component or redirects to login instead of returning Route

ayaanqui committed:
fix: remove redundant Route import

ayaanqui committed:
fix: AddPage.test.ts define useNavigate

ayaanqui committed:
fix: declare useNavigate on SubmitButton.test.ts

ayaanqui committed:
fix: remove beforeEach + define useNavigate

ayaanqui committed:
fix: update EditInfoPage.test.tsx to work with react-router v6

ayaanqui committed:
fix: remove BrowserRouter from PhenotypeContainer.test.tsx

ayaanqui committed:
test: define useNavigate for CatalogWrapper.test.tsx

ayaanqui committed:
test: remove Router wrapper for Information.test.tsx

ayaanqui committed:
test: define useNavigate to replace useHistory with push method for OauthSignHandler.test.ts

ayaanqui committed:
test: remove buggy test for OrderConfirmation.test.tsx

ayaanqui committed:
refactor: ensure id is not undefined

ayaanqui committed:
refactor: move common components to function

ayaanqui committed:
refactor: remove unsed react-router imports

ayaanqui committed:
refactor: add all strain and plasmid routes to common function

ayaanqui committed:
refactor: RenderRoutes for code climate

ayaanqui committed:
refactor: revert RenderRoute strain and plasmids refactor

ayaanqui committed:
refactor: simplify usePrams.id usage

ayaanqui committed:
fix: make id constant

ayaanqui committed:
fix: remove stock type from history push if filter query is not provided

ayaanqui committed:
fix: wrap history push with filter in useEffect

ayaanqui committed:
fix: ignore useEffect dependencies

ayaanqui committed:
fix: replace OrderConfirmation props with useLocation to handle submit confirmation

ayaanqui committed:
fix: use new OrderConfirmation component without props

ayaanqui committed:
feat: create OrderContext with types

ayaanqui committed:
feat: wrap with OrderProvider

ayaanqui committed:
feat: export OrderActionType as non-type

ayaanqui committed:
feat: create useOrderStore hook

ayaanqui committed:
feat: make orderID required

ayaanqui committed:
feat: set order state when user submits

ayaanqui committed:
feat: make orderID required

ayaanqui committed:
feat: remove OrderProvider wrapper

ayaanqui committed:
feat: add routing component for /order with OrderProvider

ayaanqui committed:
feat: configure for /order/submitted/:orderId

ayaanqui committed:
fix: consider undefined state for OrderContext

ayaanqui committed:
fix: add OrderProvider to test

ayaanqui committed:
fix: OrderConfirmation test to return useParams and useOrderStore

ayaanqui committed:
refactor: remove MockOrderContext component

ayaanqui committed:
refactor: add explicit variable name to ignore code climate errors

ayaanqui committed:
fix: update visual testing to account for /submitted/:orderId route

ayaanqui committed:
refactor: code climate similar blocks issue

ayaanqui committed:
test: add useOrderStore tests

ayaanqui committed:
test: define Navigate component

ayaanqui committed:
test: add mock function

ayaanqui committed:
test: OrderConfirmation 100% coverage

ayaanqui committed:
refactor: remove redundant switch statement

ayaanqui committed:
test: full coverage on PaymentPage

ayaanqui committed:
test: full coverage on SubmitError

mergify[bot] committed:
Merge branch 'develop' into feat/catalog-search-fields

ayaanqui committed:
test: add test cases for useGoogleAnalytics

ayaanqui committed:
refactor: smaller chunks for codeclimate

ayaanqui committed:
refactor: simplify state conditional for codeclimate

mergify[bot] committed:
Merge branch 'develop' into feat/catalog-search-fields

ayaanqui committed:
refactor: simplify PDF component for codeclimate

cybersiddhu committed:
fix: fix issues with code came from conflict resolution

cybersiddhu committed:
chore: change node version requirement to v16
